cmake/OpenCVFindOpenEXR.cmake

```diff
@@ -144,6 +144,11 @@ IF(OPENEXR_FOUND)
   endif()
   if(NOT OPENEXR_VERSION)
     SET(OPENEXR_VERSION "Unknown")
+  else()
+    if(HAVE_CXX17 AND OPENEXR_VERSION STRLESS "2_3")
```

Comment:
> `STRLESS`
> `"2_3"`

Why is not `VERSION_LESS`?

> OPENEXR_VERSION=2.3.0

Reply:
Thank you for your comment! 

I think usually ‘VERSION_LESS` is better to compare version strings, but OpenCV use not standard version string for system OpenEXR, which is splited with "_" instead of ".". 

Like built-in OpenEXR. I'll try to fix this at this pull request.

https://github.com/opencv/opencv/blob/1b48eafe48a29ddea22a47601471d7487b7ae636/cmake/OpenCVFindOpenEXR.cmake#L100-L102

https://cmake.org/cmake/help/latest/command/if.html#version-comparisons

> if(<variable|string> VERSION_LESS <variable|string>)
> Component-wise integer version number comparison (version format is major[.minor[.patch[.tweak]]], omitted components are treated as zero).

Build-in OpenEXR (Ver 2.3.0)
```
--     PNG:                         /usr/lib/x86_64-linux-gnu/libpng.so (ver 1.6.43)
--     TIFF:                        /usr/lib/x86_64-linux-gnu/libtiff.so (ver 42 / 4.5.1)
--     JPEG 2000:                   OpenJPEG (ver 2.5.0)
--     OpenEXR:                     build (ver 2.3.0)
```

System OpenEXR (Ver 2_2)
```
--     PNG:                         /usr/lib/x86_64-linux-gnu/libpng.so (ver 1.6.43)
--     TIFF:                        /usr/lib/x86_64-linux-gnu/libtiff.so (ver 42 / 4.5.1)
--     JPEG 2000:                   OpenJPEG (ver 2.5.0)
--     OpenEXR:                     /usr/local/lib/libImath-2_2.so /usr/local/lib/libIlmImf-2_2.so /usr/local/lib/libIex-2_2.so /usr/local/lib/libHalf.so /usr/local/lib/libIlmThread-2_2.so (ver 2_2)
```

Reply:
I fixed it. Currently `OPENEXR_VERSION` had been also used to search libraries.
So I split it into `OPENEXR_VERSION_MM` to search and `OPENEXR_VERSION` to show and comparing.

```
--     PNG:                         /usr/lib/x86_64-linux-gnu/libpng.so (ver 1.6.43)
--     TIFF:                        /usr/lib/x86_64-linux-gnu/libtiff.so (ver 42 / 4.5.1)
--     JPEG 2000:                   OpenJPEG (ver 2.5.0)
--     OpenEXR:                     /usr/local/lib/libImath-2_2.so /usr/local/lib/libIlmImf-2_2.so /usr/local/lib/libIex-2_2.so /usr/local/lib/libHalf.so /usr/local/lib/libIlmThread-2_2.so (ver 2.2.0)
```
And I also fixed potential problem.

```
    IF (OPENEXR_VERSION_MAJOR AND OPENEXR_VERSION_MINOR)
        set(OPENEXR_VERSION "${OPENEXR_VERSION_MAJOR}_${OPENEXR_VERSION_MINOR}")
    ENDIF ()
```

If OpenEXR Ver x.0.y is used, this `set(OPENEXR_VERSION)` statement will not be executed.

```
    IF (OPENEXR_VERSION_MAJOR AND OPENEXR_VERSION_MINOR)

    IF (4 AND 0)

    IF (TRUE AND FALSE)  # non-zero means TRUE, zero means FALSE

    IF (FALSE)
```